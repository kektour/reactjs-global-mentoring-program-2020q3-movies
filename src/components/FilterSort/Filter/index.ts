export { Filter } from './Filter';
export type { Item } from './Filter';
