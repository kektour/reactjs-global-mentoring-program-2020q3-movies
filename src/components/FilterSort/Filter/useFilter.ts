import { useDispatch } from 'react-redux';
import { moviesSetGenreFilterActionCreator, MoviesSetGenreFilterAction } from '../../../store/movies';

type UseFilter = {
  filterMovies: (genre: string) => MoviesSetGenreFilterAction;
};

export const useFilter = (): UseFilter => {
  const dispatch = useDispatch();
  const filterMovies = (genre: string) => dispatch(moviesSetGenreFilterActionCreator(genre));

  return {
    filterMovies,
  };
};
