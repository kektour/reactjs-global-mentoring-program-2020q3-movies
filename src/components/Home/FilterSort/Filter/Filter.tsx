import React, { useCallback, useState } from 'react';
import { FilterButton } from './FilterButton';
import styles from './Filter.module.scss';
import { useFilter } from './useFilter';

export interface Item {
  id: string;
  name: string;
}

type Props = {};

const items: Array<Item> = [
  {
    id: 'all',
    name: 'All',
  },
  {
    id: 'documentary',
    name: 'Documentary',
  },
  {
    id: 'comedy',
    name: 'Comedy',
  },
  {
    id: 'horror',
    name: 'Horror',
  },
  {
    id: 'crime',
    name: 'Crime',
  },
];

export const Filter: React.FC<Props> = () => {
  const [item, setItem] = useState(items[0].id);
  const filter = useFilter();
  const handleFilterButtonClick = useCallback((id: string) => {
    setItem(id);
    filter.filterMovies(id);
  }, [filter]);

  return (
    <div className={styles.root}>
      {items.map((i) => (
        <FilterButton key={i.id} item={i} isSelected={i.id === item} onClick={handleFilterButtonClick} />
      ))}
    </div>
  );
};
