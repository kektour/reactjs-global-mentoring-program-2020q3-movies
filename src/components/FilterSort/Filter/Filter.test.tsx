import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { Filter } from './Filter';

const mockStore = configureStore();

describe('Filter', () => {
  it('Should render component correctly', () => {
    const store = mockStore({});
    const component = render(
      <Provider store={store}>
        <Filter />
      </Provider>
    );
    expect(component.asFragment()).toMatchSnapshot();
  });
});
