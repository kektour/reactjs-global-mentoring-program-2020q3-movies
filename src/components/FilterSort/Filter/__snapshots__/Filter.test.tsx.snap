// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Should render component correctly 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <div
      class="root"
    >
      <button
        class="button buttonActive"
      >
        All
      </button>
      <div
        class="line"
      />
    </div>
    <div
      class="root"
    >
      <button
        class="button"
      >
        Documentary
      </button>
    </div>
    <div
      class="root"
    >
      <button
        class="button"
      >
        Comedy
      </button>
    </div>
    <div
      class="root"
    >
      <button
        class="button"
      >
        Horror
      </button>
    </div>
    <div
      class="root"
    >
      <button
        class="button"
      >
        Crime
      </button>
    </div>
  </div>
</DocumentFragment>
`;
